src/lints/manual/manual_unwrap_or_else.rs

```diff
@@ -0,0 +1,301 @@
+/// struct Struct {
+///     x: felt252
+/// }
+///
+/// let foo: Option<Struct> = None;
+/// match foo {
+///     Some(v) => v,
+///     None => Struct { x: 0x0 },
+/// };
+/// ```
+///
+/// Can be simplified to:
+///
+/// ```cairo
+/// let foo: Option<i32> = None;
+/// foo.unwrap_or_else(|| Struct { x: 0x0 });
+/// ```
+impl Lint for ManualUnwrapOrElse {
+    fn allowed_name(&self) -> &'static str {
+        "manual_unwrap_or_else"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "Manual `unwrap_or_else` detected. Consider using `unwrap_or_else()` instead."
+    }
+
+    fn kind(&self) -> CairoLintKind {
+        CairoLintKind::ManualUnwrapOrElse
+    }
+
+    fn has_fixer(&self) -> bool {
+        true
+    }
+
+    fn fix<'db>(&self, db: &'db dyn Database, node: SyntaxNode<'db>) -> Option<InternalFix<'db>> {
+        fix_manual_unwrap_or_else(db, node)
+    }
+
+    fn fix_message(&self) -> Option<&'static str> {
+        Some("Use `unwrap_or_else()` instead of manual pattern")
+    }
+}
+
+#[tracing::instrument(skip_all, level = "trace")]
+pub fn check_manual_unwrap_or_else<'db>(
+    db: &'db dyn Database,
+    item: &ModuleItemId<'db>,
+    diagnostics: &mut Vec<PluginDiagnostic<'db>>,
+) {
+    let function_bodies = get_all_function_bodies_with_ids(db, item);
+
+    for (function_id, function_body) in function_bodies {
+        let if_exprs = get_all_if_expressions(function_body);
+        let match_exprs = get_all_match_expressions(function_body);
+        let arenas = &function_body.arenas;
+
+        for match_expr in match_exprs {
+            if check_manual_unwrap_or_else_with_match(db, &match_expr, function_id, arenas) {
+                diagnostics.push(PluginDiagnostic {
+                    stable_ptr: match_expr.stable_ptr.untyped(),
+                    message: ManualUnwrapOrElse.diagnostic_message().to_owned(),
+                    severity: Severity::Warning,
+                    inner_span: None,
+                });
+            }
+        }
+
+        for if_expr in if_exprs {
+            if check_manual_unwrap_or_else_with_if(db, &if_expr, function_id, arenas) {
+                diagnostics.push(PluginDiagnostic {
+                    stable_ptr: if_expr.stable_ptr.untyped(),
+                    message: ManualUnwrapOrElse.diagnostic_message().to_owned(),
+                    severity: Severity::Warning,
+                    inner_span: None,
+                });
+            }
+        }
+    }
+}
+
+fn check_manual_unwrap_or_else_with_match<'db>(
+    db: &'db dyn Database,
+    match_expr: &ExprMatch<'db>,
+    function_id: FunctionWithBodyId<'db>,
+    arenas: &Arenas<'db>,
+) -> bool {
+    let matched_expr = db.expr_semantic(function_id, match_expr.matched_expr);
+    let is_droppable = db.droppable(matched_expr.ty()).is_ok();
+    let is_manual_unwrap_or_else =
+        check_manual(db, match_expr, arenas, ManualLint::ManualUnwrapOrElse);
+    is_manual_unwrap_or_else && !is_droppable
```

Comment:
```suggestion
    let is_droppable = db.droppable(matched_expr.ty()).is_ok();
    
    !is_droppable && check_manual(db, match_expr, arenas, ManualLint::ManualUnwrapOrElse)
```

Nit: this way we don't execute `check_manual` if it is not necessary, a little bit faster